SPECIFICATION.md

```diff
@@ -0,0 +1,293 @@
+# Playbook specification
+
+This document describes the exact format required by the Ansible playbook verifier.
+
+```yaml
+---
+- name: Example playbook
+  hosts: localhost
+  become: yes
+  vars:
+    insights_signature_exclude: /hosts,/vars/insights_signature
+    insights_signature: !!binary |
+      base64 binary blob containing the embedded GPG signature
+  tasks:
+    - name: Display debug message
+    - ansible.builtin.debug:
+        msg: The signed playbook says hello!
+```
+
+## Only one playbook may be present
+
+In case there are multiple playbooks present in the file, only the first one will be verified. Any other playbook is explicitly ignored.
+
+**Warning:** Even though only the first playbook is verified, all of them may be emitted to standard output after the verification.
+
+<details>
+
+<summary>Example</summary>
+
+```yaml
+# valid
+- name: Example playbook
+  hosts: localhost
+  ...
+```
+```yaml
+# invalid
+- name: Example playbook
+  hosts: localhost
+  ...
+- name: Another playbook
+  hosts: localhost
+  ...
+```
+
+</details>
+
+## Booleans
+
+YAML parsers may differ in how they interpret all the booleans the specification allows.
+
+The following values MUST evaluate to boolean: `true`, `True`, `TRUE`.
+
+The following values MUST stay strings: `y`, `Y`, `yes`, `Yes`, `YES`, `on`, `On`, `ON`.
+
+When serialized, `true` must be formatted as `True`.
+
+## Unquoted YAML aliases
+
+The document MUST NOT contain string-like values that could parse as aliases.
+
+<details>
+
+<summary>Example</summary>
+
+```yaml
+serve:
+  - /robots.txt
+  - /favicon.ico
+  - *.html
+```
+
+While it may parse correctly in some libraries, the reference implementation errors out:
+
+```py
+found undefined alias '.html'
+  in "<unicode string>", line 15, column 9:
+    - *.html
+      ^ (line: 15)
+```
+
+</details>
+
+## YAML tags
+
+YAML specification defines tags, denoted by `!` prefix. While the tags get parsed by the reference implementation without raising an error, their serialization is not static and they MUST not be used.
+
+<details>
+
+<summary>Example of mismanaged serialization</summary>
+
+```yaml
+serve:
+  - /robots.txt
+  - !.git
+```
+```py
+('serve', ['/robots.txt', <insights.client.apps.ansible.playbook_verifier.contrib.ruamel_yaml.ruamel.yaml.comments.TaggedScalar object at 0x7fd9ad7ebdc0>])
+```
+
+</details>
+
+## Unicode
+
+Unicode support exists since RHEL 8. In RHEL 7, Python 2.7 is used, which ignores all non-ASCII characters.
+
+If the playbook targets RHEL 7, characters outside of ASCII range MUST NOT be used.
+
+For systems targeting RHEL 8 and over, characters outside of the ASCII range MUST be encoded as `\x00`. <!-- TODO: mention endianness -->
```

Comment:
rather than mention RHEL, i'd only mention the version of Python, i.e. Python 2 vs 3